HARK/ConsumptionSaving/ConsMedModel.py

```diff
@@ -714,7 +714,31 @@ def updatepLvlGrid(self):
         for j in range(len(self.pLvlGrid)): # Then add 0 to the bottom of each pLvlGrid
             this_grid = self.pLvlGrid[j]
             self.pLvlGrid[j] = np.insert(this_grid,0,0.0001)
-
+            
+    def resetRNG(self):
+        '''
+        Reset the RNG behavior of this type.  This method is called automatically
+        by initializeSim(), ensuring that each simulation run uses the same sequence
+        of random shocks; this is necessary for structural estimation to work.
+        This method extends PersistentShockConsumerType.resetRNG() to also reset
+        elements of MedShkDstn.
+        
+        Parameters
+        ----------
+        None
+        
+        Returns
+        -------
+        None
+        '''
+        PersistentShockConsumerType.resetRNG(self)
+        
+        # Reset MedShkDstn if it exists (it might not because resetRNG is called at init)
```

Comment:
Again: if all the exogenous shock distributions were contained in a single collector, then this method could be written more efficiently by looping over that collection.